# Correlation Analysis Project Code


In [2]:
import csv
import math
import statistics as stats
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt


mcas = pd.read_csv('MCAS.csv') #read in file to dataframe

mcas[:3]



#Data Processing - Clean up and split the data into two lists, east and east
#This allows for further statistical analysis on cleaned up lists
#this is traceys code

#remove the first element (column header)
#del MCAS[0];

#print(MCAS)

print("regday r    =", np.corrcoef(mcas['totsc4'], mcas['regday'])[0, 1])
print("specneed r  =", np.corrcoef(mcas['totsc4'], mcas['specneed'])[0, 1])
print("bilingua r  =", np.corrcoef(mcas['totsc4'], mcas['bilingua'])[0, 1])
print("occupday r  =", np.corrcoef(mcas['totsc4'], mcas['occupday'])[0, 1])
print("totday r    =", np.corrcoef(mcas['totsc4'], mcas['totday'])[0, 1])
print("spc r       =", np.corrcoef(mcas['totsc4'], mcas['spc'])[0, 1])
print("speced r    =", np.corrcoef(mcas['totsc4'], mcas['speced'])[0, 1])
print("lnchpct r   =", np.corrcoef(mcas['totsc4'], mcas['lnchpct'])[0, 1])
print("tchratio r  =", np.corrcoef(mcas['totsc4'], mcas['tchratio'])[0, 1])
print("percap r    =", np.corrcoef(mcas['totsc4'], mcas['percap'])[0, 1])
print("totsc8 r    =", np.corrcoef(mcas['totsc4'], mcas['totsc8'])[0, 1])
print("avgsalary r =", np.corrcoef(mcas['totsc4'], mcas['avgsalary'])[0, 1])
print("pctel r     =", np.corrcoef(mcas['totsc4'], mcas['pctel'])[0, 1])

regdayR    = np.corrcoef(mcas['totsc4'], mcas['regday'])[0, 1]
specneedR  = np.corrcoef(mcas['totsc4'], mcas['specneed'])[0, 1]
bilinguaR  = np.corrcoef(mcas['totsc4'], mcas['bilingua'])[0, 1]
occupdayR  = np.corrcoef(mcas['totsc4'], mcas['occupday'])[0, 1]
totdayR    = np.corrcoef(mcas['totsc4'], mcas['totday'])[0, 1]
spcR       = np.corrcoef(mcas['totsc4'], mcas['spc'])[0, 1]
specedR    = np.corrcoef(mcas['totsc4'], mcas['speced'])[0, 1]
lnchpctR   = np.corrcoef(mcas['totsc4'], mcas['lnchpct'])[0, 1]
tchratioR  = np.corrcoef(mcas['totsc4'], mcas['tchratio'])[0, 1]
percapR    = np.corrcoef(mcas['totsc4'], mcas['percap'])[0, 1]
totsc8R    = np.corrcoef(mcas['totsc4'], mcas['totsc8'])[0, 1]
avgsalaryR = np.corrcoef(mcas['totsc4'], mcas['avgsalary'])[0, 1]
pctelR     = np.corrcoef(mcas['totsc4'], mcas['pctel'])[0, 1]

#create a list of correlation coeffcients for minutes 0 - 47
r_vals = [] 

#loop through columns and compute Pearson correlation coefficient
for c in mcas.columns:
    #skip the team name column and the minute 48 column (its our anchor variable)
    if c != 'Column1' and c != 'code' and c != 'municipa' and c != 'district' and c != 'totsc4' and c != 'spc' and c != 'totsc8' and c != 'avgsalary':
        #calculate the coefficient and append it to our list
        r = np.corrcoef(mcas[c], mcas['totsc4'])[0, 1]
        r_vals.append(r) 
        print("Correlation in ", c, " is ", r)
        
        

# make sure figures appear inline in Ipython Notebook
#%matplotlib inline
# create a quick bar chart
#mcas['totsc4'].plot(kind='bar');
#plt.plot(kind='bar');
print()
print(r_vals)

plt.xlabel("Minute in game")
plt.ylabel("R correlation to final minute")
plt.plot('r_vals')
plt.show()




regday r    = 0.179493063031
specneed r  = 0.0931881628459
bilingua r  = -0.100864012695
occupday r  = -0.418619464134
totday r    = 0.108952042221
spc r       = nan
speced r    = -0.241002372748
lnchpct r   = -0.78418370215
tchratio r  = -0.258545334661
percap r    = 0.623420563566
totsc8 r    = nan
avgsalary r = nan
pctel r     = -0.52797467598
Correlation in  Unnamed: 0  is  0.0187515854994
Correlation in  regday  is  0.179493063031
Correlation in  specneed  is  0.0931881628459
Correlation in  bilingua  is  -0.100864012695
Correlation in  occupday  is  -0.418619464134
Correlation in  totday  is  0.108952042221
Correlation in  speced  is  -0.241002372748
Correlation in  lnchpct  is  -0.78418370215
Correlation in  tchratio  is  -0.258545334661
Correlation in  percap  is  0.623420563566
Correlation in  pctel  is  -0.52797467598

[0.018751585499391845, 0.1794930630310102, 0.093188162845920075, -0.10086401269458166, -0.41861946413446927, 0.10895204222089605, -0.24100237274760256, -0.7841

ValueError: could not convert string to float: 'r_vals'

# Comentary 
Three most corelated to totsc4 are 
lnchpct r   = -0.78418370215
percap r    = 0.623420563566
pctel r     = -0.52797467598

The three most corelated data sets to 4th grade score, (not including the sets that weren't complete) where negatively "% Eligible for free/reduced price lunch", positively "Per Capita Income", and negatively "Percent English Learners". Lower income often results in higher more free/reduced lunch, because its set up, and then the possible causation could be that people who are less wealthy don’t have the means to as good of an education. 